thunder/core/transforms.py

```diff
@@ -1357,6 +1357,28 @@ def _embedding_prim_grad(
 
 register_grad(pids.EMBEDDING, _embedding_prim_grad)
 
+
+def _maximum_grad(a: TensorProxy, b: TensorProxy, /):
+    fwd = prims.maximum(a, b)
+
+    g = get_grad(fwd)
+
+    # Compute sub-gradient if `a == b`
+    a_grad = prims.where(a == b, g / 2, 0.0)
+    b_grad = prims.where(a == b, g / 2, 0.0)
```

Comment:
If `g` is `NaN` somewhere, then what should the gradient for `a` and `b` be in that location?

Reply:
If `NaN` is somewhere in `g`, then it would just propagate depending on whether to which tensor had the max value or in case the values were equal then it would propagate through gradients of both tensors. I dont think there is much that can be done if NaN is already present in `g`

```python
>>> t = torch.tensor(1., requires_grad=True)
>>> s = torch.tensor(2., requires_grad=True)
>>> torch.max(t, s).backward(torch.tensor(float("nan")))
>>> t.grad
tensor(0.)
>>> s.grad
tensor(nan)
>>> 
>>> t = torch.tensor(1., requires_grad=True)
>>> s = torch.tensor(1., requires_grad=True)
>>> torch.max(t, s).backward(torch.tensor(float("nan")))
>>> t.grad
tensor(nan)
>>> s.grad
tensor(nan)
```

Reply:
Sounds great -- would you just add that to the comment and explain how the logic enforces this?